compiler/z/codegen/ControlFlowEvaluator.cpp

```diff
@@ -1790,29 +1790,29 @@ OMR::Z::TreeEvaluator::tableEvaluator(TR::Node * node, TR::CodeGenerator * cg)
       case AddressTable64bitLongLookup:
          {
          if(!node->chkCannotOverflow())
-           generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::CLG, node, selectorReg, static_cast<int64_t>(numBranchTableEntries), TR::InstOpCode::COND_BNL, node->getSecondChild()->getBranchDestination()->getNode()->getLabel(), false, false);
+           generateS390CompareAndBranchInstruction(cg, TR::InstOpCode::CLG, node, selectorReg, static_cast<int32_t>(numBranchTableEntries), TR::InstOpCode::COND_BNL, node->getSecondChild()->getBranchDestination()->getNode()->getLabel(), false, false);
```

Comment:
I do not expect `numBranchTableEntries` to be larger (Given that it is number of entries in the Branch table) But as it is declared as `uint32_t` and `selectorReg` is 64Bit  We should for sanity use 64 Bit int here.